src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -605,6 +607,23 @@ void Graph::AllocateWithReuse() {
                 erase = true;
                 break;
             }
+
+            // Special allocation for string tensors
+            if (edge->getDesc().getPrecision() == element::string && edge->getStatus() == Edge::Status::NeedAllocation &&
+                    !edge->getParent()->isConstant()) {
+                auto memory = std::make_shared<StringMemory>(getEngine(), edge->getDesc());
+                edge->reuse(memory);
+                for (auto& edge_c : cluster) {
+                    if (edge_c->getDesc().getPrecision() == element::string && edge_c->getStatus() == Edge::Status::NeedAllocation &&
+                            !edge_c->getParent()->isConstant()) {
```

Comment:
What about a constant path? `isConstant()` is not a check for a node type, but for the situation when all the tensors attached to the node are constant tensors. Following this code, string tensor on constant paths other than directly attached to nodes of type `Constant` are allocated in the mainstream flow, which is not we expect to happen. Thus, it looks like `!edge_c->getParent()->isConstant()` should be replaced with !(edge_c->getParent()->isConstant() && edge_c->getParent()->getType() == Type::Input). So that to skip only `op::Constant` nodes.

Reply:
Done